pkg/pulumiyaml/ast/template.go

```diff
@@ -423,23 +426,30 @@ func (d *ResourceDecl) recordSyntax() *syntax.Node {
 
 // The names of exported fields.
 func (*ResourceDecl) Fields() []string {
-	return []string{"type", "defaultprovider", "properties", "options", "get"}
+	return []string{"type", "logicalname", "defaultprovider", "properties", "options", "get"}
```

Comment:
I wonder if the user facing name for this should just be "name"?

Reply:
Is this a very common use-case? 🤔 YAML already allows snake-case names and other weird characters in the key of the resource. Though `name` does make sense here as well.

Reply:
Not really but in all the other sdks we just call this ctor parameter "name". I think it would be very consistent to do that here.

Reply:
Alright, will change it to `name` 👍 same for `config` blocks

Reply:
Done ✅